src/universal-app/kitchen-sink/kitchen-sink.html

```diff
@@ -257,18 +257,23 @@ <h2>Select</h2>
 <h2>Sidenav</h2>
 <mat-sidenav-container>
   <!--
-    TODO: add sample with multiple sidenavs which can be problematic for hydration.
-    Currently blocked on https://github.com/angular/angular/issues/53246
+    Place the end sidenav first to test the behavior where we manually change the DOM order.
   -->
+  <mat-sidenav #endSidenav position="end" mode="push">
+    End sidenav
+    <button (click)="endSidenav.toggle()">Close</button>
+  </mat-sidenav>
+
   <!--
     Only attempt to capture focus when automated, otherwise it makes the page jump around.
    -->
-  <mat-sidenav #sidenav opened [autoFocus]="isAutomated ? 'first-tabbable' : false">
-    On the side
-    <button (click)="sidenav.toggle()">Button for testing focus trapping</button>
+  <mat-sidenav #startSidenav opened [autoFocus]="isAutomated ? 'first-tabbable' : false">
+    Start sidenav
+    <button (click)="startSidenav.toggle()">Button for testing focus trapping</button>
   </mat-sidenav>
   Main content
-  <button (click)="sidenav.toggle()">Click me</button>
+  <button (click)="startSidenav.toggle()">Toggle start</button>
+  <button (click)="endSidenav.toggle()">Toggle end</button>
 </mat-sidenav-container>
 
 <h2>Slide-toggle</h2>
```

Comment:
BTW: One thing I did play with in the past was to use Pupeteer and capture a screenshot of the SSR structure to ensure the initial "boilerplate" looks as expected. I wonder, do we also bother about this with these PRs or are we just ensuring no errors?

Reply:
We're just ensuring that there's no errors, but hydration errors actually aren't captured on the CI yet. Eventually it would be nice to add something (potentially Puppeteer or Webdriver) that opens the client-side-rendered page and checks that there are no JS errors.